[Distributed] Add finer granularity tag for distributed submodule

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116434

This PR is the start to enable the integrate pytorch distributed logs in Torch LOGs. We now already have one tag "distributed" for all distributed components but distributed is a very large component and we want to have some hierarchy and give users options to only turn on logs for certain submodules. So we also added tags starting with "dist_*" for each submodule. (This PR only adds some of them and we are going to add more down the road)

Related discussions can be found here: https://github.com/pytorch/pytorch/issues/113544

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @wz337 @tianyu-l @wconstab @yf225 @fritzo @neerajprad @alicanb @nikitaved

fduwjj committed:
[Distributed] Add finer granularity tag for distributed submodule

[ghstack-poisoned]

awgu commented on torch/_logging/_internal.py:
```diff
@@ -254,6 +257,18 @@ def set_logs(
             Default: ``logging.WARN``
 
+        dist_c10d (:class:`Optional[int]`):
+            Whether to log communication operations related debug info in pytorch distributed components.
```
nit (before this PR): Should we stylize as "PyTorch Distributed"?

fduwjj replied:
Agree... Actually, PyTorch part is for sure, I am not sure if we should capitalize distributed. Or we just treat "PyTorch Distributed" as a term?

fduwjj replied:
Actually by checking PTD's code base, looks like we already using `PyTorch Distributed`, let's just keep this tradition. (Maybe change all PyTorch distributed to PyTorch Distributed as a follow-up)

awgu reviewed (approved):
This looks reasonable to me!

wanchaol commented on torch/_logging/_internal.py:
```diff
@@ -254,6 +257,18 @@ def set_logs(
             Default: ``logging.WARN``
 
+        dist_c10d (:class:`Optional[int]`):
+            Whether to log communication operations related debug info in pytorch distributed components.
```
nit: whether to log c10d communication operation

fduwjj replied:
Actually, by consulting with ChatGPT, it says:

> Use "operations" if you are referring to multiple actions or procedures, or to the general concept of a series of actions or procedures.

Here we actually want loggings for multiple actions (collectives, rendezvous), so that we keep it same as L257. Let me know if this sounds make sense or not.

wanchaol replied:
oh I meant adding "c10d", not commenting about the "operation" :)

fduwjj replied:
oh, sorry. Missed that, let me send out an another PR for this.

wanchaol commented on torch/_logging/_registrations.py:
```diff
@@ -2,7 +2,11 @@
+DISTRIBUTED = [
+    "torch.distributed",
+    "torch._dynamo.backends.distributed",
+    "torch.nn.parallel.distributed",
```
missing fsdp here?

fduwjj replied:
Oh, fsdp is a child of "torch.distributed", so it should be in already.

wanchaol commented on torch/_logging/_registrations.py:
```diff
@@ -11,6 +15,11 @@
+    "dist_c10d", ["torch.distributed.distributed_c10d", "torch.distributed.rendezvous"]
+)
+register_log("dist_ddp", ["torch.nn.parallel.distributed"])
+register_log("dist_fsdp", ["torch.distributed.fsdp"])
```
I hope we won't add too many top level entries in the future that make TORCH_LOGS hard to use, if that happens we can consider a hierarchical parsing like `distributed.ddp` at that time I guess.

fduwjj replied:
Agree, that's why I didn't add so many here. We should at most have 5 or 6 submodule labels.. The current three should cover most debugging experiences IMHO.

wanchaol reviewed (approved):
lgtm! some nits only

fduwjj committed:
Update on "[Distributed] Add finer granularity tag for distributed submodule"


This PR is the start to enable the integrate pytorch distributed logs in Torch LOGs. We now already have one tag "distributed" for all distributed components but distributed is a very large component and we want to have some hierarchy and give users options to only turn on logs for certain submodules. So we also added tags starting with "dist_*" for each submodule. (This PR only adds some of them and we are going to add more down the road)

Related discussions can be found here: https://github.com/pytorch/pytorch/issues/113544

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol wz337 tianyu-l wconstab yf225 fritzo neerajprad alicanb nikitaved

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge
